### PR TITLE
feat: swap via paraswap

### PR DESCRIPTION
**Pull request by saledjenic**
Swap using paraswap is integrated on the status-go side.

Tested in the desktop app, unfortunately couldn't test it on the testnet, cause it's not supported by the paraswap. 

All swaps made via the desktop app can be seen here https://optimistic.etherscan.io/address/0x946a89180365d054677d10b1a73235c0bee6734f cause that's the address with some real assets made for testing. 

In the desktop app a swap ETH to ERC20 was tested using using password and keycard and both worked.

A swap ERC20 to ETH works, but partially, but on every second try. :)
The thing is that for this kind of swap paraswap in order to build a transaction via paraswaop, it requires an approval transaction not only to be placed, but executed. Since we're placing an approval transaction first and immediately after it a swap transaction, the swap transaction fails, cause the approval transaction is not executed yet. Trying the same again will do the swap, cause an approval is already executed.

That's something we need to handle either delaying an execution of swap transaction or separating approval and swap transaction, it's up to the UI and how designers want to see that feature. 

Till then we're free to merge this since it doesn't break anything and once we have the design ready, we can just adjust the change based on final requirements.

### CODE REVIEW

File: services/wallet/bridge/swap_paraswap.go
Comment (alaibe) [inline services/wallet/bridge/swap_paraswap.go:59]: should this be cached for some time? i don't see that changing every minute.

I guess the client should do caching right?

File: services/wallet/bridge/swap_paraswap.go
Comment (alaibe) [inline services/wallet/bridge/swap_paraswap.go:91]: There is no fees from paraswap itself, like a provider fee?

File: services/wallet/router.go
Comment (alaibe) [inline services/wallet/router.go]: in case of optimism, we still need this right?

File: services/wallet/bridge/swap_paraswap.go
Comment (saledjenic) [reply services/wallet/bridge/swap_paraswap.go:59]: We don't do this always, only when the token has no address set, which should be no case or a rare case.

File: services/wallet/bridge/swap_paraswap.go
Comment (saledjenic) [reply services/wallet/bridge/swap_paraswap.go:91]: Not sure, this is from their documentation: https://developers.paraswap.network/smart-contracts/augustus-swapper#fee-structure

`ParaSwap currently doesn't charge any fees from its end-users directly, except in the case of positive slippage.`

File: services/wallet/router.go
Comment (saledjenic) [reply services/wallet/router.go]: Also not sure about it, most likely yes, but I added that condition because of an inability to build a tx for swapping erc20 tokens without executed approval tx.

Comment (saledjenic) [reply services/wallet/router.go]: Since we're going to check and test the entire logic once we have the final client use cases I removed `sendType != Swap` part from the condition.

File: services/wallet/bridge/swap_paraswap.go
Comment (alaibe) [reply services/wallet/bridge/swap_paraswap.go:91]: I guess the what you will receive is also explicit about fees

**Review by alaibe — APPROVED** [review]

**Comment by saledjenic** [comment]
@shivekkhurana I am not sure if the mobile team is using this `GetSuggestedRoutes` or not, but if yes, could you please add `toTokenID` param to the relevant call in the mobile app and you can set it to an empty string till the moment you start swap implementation.

**Comment by J-Son89** [comment]
@saledjenic, @Samyoul, @cammellos, @iurimatias @churik  -
Here we are knowingly introducing a breaking change to an external API of Status-Go.

Do we have a clear strategy for how we do this, should mobile not test this works fine before merging??

Additionally we should probably have some format to make this as clear as possible, e.g the title should include breaking-changes or we have a label associated with it.

**Pull request merged**
